core/embed/rust/src/translations/generated/translated_string.rs.mako

```diff
@@ -55,7 +56,7 @@ impl TranslatedString {
             %if any(name.startswith(prefix + "__") for prefix in ALTCOIN_PREFIXES):
             #[cfg(feature = "universal_fw")]
             %endif
-            Self::${name} => ${json.dumps(en_data.get(name, '""'))},
+            Self::${name} => ${re.sub(r'\\u([0-9a-f]{1,5})', r'\\u{\g<1>}', json.dumps(en_data.get(name, '""')))},
```

Comment:
why is this here?

Reply:
Rust requires `\u{NNN}` rather than `\uNNN` which is what JSON yields.

Reply:
That being said, @matejcik - I found one issue with my regexp... So I decided to change it to something more conservative: https://github.com/trezor/trezor-firmware/pull/4010/commits/9ad8417c16f96240a869de880317b9823db5f7f5#diff-6b77a3cb1462becb160af081a8eb77d7c3f1c77d4365438e622b4ea44c139e6cR59

Issue was... if the word after the non-breaking space would start with a-f ... the first letter(s) of that word could be considered to be part of the code (matched by the regexp) and yield an invalid character.

So, my new solution is, rather than trying to match anything, just match `00a0` ... Anything else would lead to a Rust compilation error, in which case we could edit the regexp and add more cases... Although, most likely we don't really want any special characters anyway!

Reply:
yes, well, per json.org the escape must be exactly 4 hex chars in all cases:
https://www.json.org/img/string.png
dunno why you figured the `{1,5}` thing...? (in your defence i missed it too)

Reply:
If it's always 4, it should be safe to use a regexp. Changed.